src/hooks/useCallRequest.ts

```diff
@@ -0,0 +1,23 @@
+import type { QueryFunction } from '@tanstack/react-query';
+import { useQueryClient } from '@tanstack/react-query';
+import type { SpindlFetchParams } from 'src/types/spindl';
+
+export const useCallRequest = () => {
+  const queryClient = useQueryClient();
+
+  const fetch = async (queryFn: QueryFunction, params: SpindlFetchParams) => {
```

Comment:
Should not be named fetch, that's overlapping with window.fetch and might be confusing